sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -712,15 +710,20 @@ static pi_result getOrCreatePlatform(ze_driver_handle_t ZeDriver,
     static auto PiPlatformsCacheMutex = new std::mutex;
 
     std::lock_guard<std::mutex> Lock(*PiPlatformsCacheMutex);
-    for (const pi_platform &CachedPlatform : *PiPlatformsCache) {
-      if (CachedPlatform->ZeDriver == ZeDriver) {
-        Platform[0] = CachedPlatform;
-        return PI_SUCCESS;
-      }
+    if (!PiPlatformsCache->empty()) {
+      *PlatformCache = PiPlatformsCache;
+      return PI_SUCCESS;
     }
 
-    // TODO: figure out how/when to release this memory
-    *Platform = new _pi_platform(ZeDriver);
+    uint32_t ZeDriverCount = 0;
+    ZE_CALL(zeDriverGet(&ZeDriverCount, nullptr));
+    if (ZeDriverCount == 0) {
+      return PI_SUCCESS;
```

Comment:
You need to set `*PlatformCache` before returning `PI_SUCCESS`.

Reply:
Done.

> I think it would also be good to add errors checks to the calls to `piextDeviceCreateWithNativeHandle()` and `piextPlatformCreateWithNativeHandle()`. Probably, they should throw an exception if there is an error.

Currently, error checking is already occurring  through the plugin::call interface. PI function calls (in this case 'piextDeviceCreateWithNativeHandle()' and 'piextPlatformCreateWithNativeHandle()') are passed into plugin::call as an argument and invoked from here. The result of the PI call is recorded and is passed into a function called checkPiResult() which will throw an exception if the result is not a PI_SUCCESS.

Reply:
Done.

Reply:
You also need to set `PiPlatformCachePopulated = true` here.

Reply:
I think this comment still needs to be addressed.